File: MODEL/quadrato.cpp
#include "quadrato.h"

Quadrato::Quadrato() : Quadrilatero() {}

Quadrato::Quadrato(double lato, Colore* col, QString nome) :
    Quadrilatero(lato,lato,lato,lato,Angolo(90), Angolo(90), Angolo(90), Angolo(90), col, nome){}

double Quadrato::getArea()const{
    return pow(Punto::distanceTo(Punto::origine,getCoordinate()[1]),2);
}

double Quadrato::getPerimetro()const{
    return 4*Punto::distanceTo(Punto::origine,getCoordinate()[1]);
}

void Quadrato::estendi(double fattore){
    Quadrato temp(getLati().first()*fattore);
    setPunti(temp.getCoordinate());
}

Quadrato& Quadrato::zoom(double fattore) const{
    return *(new Quadrato(getLati()[0]*fattore));
}

void Quadrato::cambiaBase(double lato){
    Quadrilatero::cambiaBase(lato);
}

Quadrato &Quadrato::specchia() const{
    QVector<Punto> vertici=getCoordinate();
    for(QVector<Punto>::iterator it=vertici.begin(); it<vertici.end(); ++it)
        it->invertiY();
    Quadrato& specchiato =*(new Quadrato(*this));
    specchiato.setPunti(vertici);
    return specchiato;
}

File: KalkJava/src/Quadrato.java
import java.util.ArrayList;

public class Quadrato extends Quadrilatero{

    public Quadrato (Double lato, Colore col, String nome){
        super(lato,lato,lato,Angolo.angoloRetto,Angolo.angoloRetto,col,nome);
    }
    public Quadrato(Double lato){
        this(lato, new RGB(), "");
    }
    public Quadrato(Quadrato quad){
        this(quad.getLati().get(0));
    }


    @Override
    public Double getArea() {
        return getLati().get(0)*getLati().get(0);
    }

    @Override
    public void estendi(Double e) {
        Quadrato quad= new Quadrato(getLati().get(0)*e);

    }

    @Override
    public Quadrato zoom(Double z) {
        return new Quadrato(getLati().get(0)*z);

    }

    @Override
    public Quadrilatero cambiaBase(Integer n) {
        return super.cambiaBase(n);
    }

    public Quadrato specchia(){
        ArrayList<Punto> vertici = getCoordinate();
        for(Punto p : vertici)
            p.invertiY();

        Quadrato specchiato = new Quadrato(this);
        specchiato.setPunti(vertici);
        return specchiato;


    }



}
